scripts/azureml-assets/azureml/assets/model/model_utils.py

```diff
@@ -151,3 +153,52 @@ def prepare_model(spec_path, model_config, registry_name, temp_dir):
     except Exception as e:
         logger.log_error(f"prepare model failed for {spec_path}. Error {e}")
         return None, False
+
+
+def update_model_metadata(
+        model_name: str, 
+        model_version: str, 
+        spec_path: Path, 
+        model_config: assets.ModelConfig,
+        registry_name: str, 
+        update: AssetVersionUpdate = None,
+    ):
+    """Update the mutable metadata of already registered Model asset."""
+    try:
+        ml_client = MLClient(credential= AzureCliCredential(), registry_name=registry_name)
+        model = ml_client.models.get(name= model_name, version= model_version)
+
+        if update is not None: ## to update older version of models
+            if update.description is not None:
+                model.description = update.description
+            if update.tags:
+                # Replace tags
+                if update.tags.replace is not None:
+                    model.tags = update.tags.replace
+                elif update.tags.add or update.tags.delete:
+                    # Add tags
+                    if update.tags.add:
+                        for k, v in update.tags.add.items():
+                            model.tags[k] = v
+
+                    # Delete tags
+                    if update.tags.delete:
+                        for k in update.tags.delete:
+                            model.tags.pop(k, None)            
+
+        else: # update the model using spec files for latest version
+            with open(spec_path) as f:
+                model_spec = YAML().load(f)
+                model.tags = model_spec["tags"]
+
+            model_description_file_path = Path(spec_path).parent / model_config.description
+            logger.print(f"model_description_file_path {model_description_file_path}")
+            if os.path.exists(model_description_file_path):
+                with open(model_description_file_path) as f:
+                    model.description = f.read()
+            else:
+                logger.print(f"description file does not exist for model {model_name}")
+
+        ml_client.models.create_or_update(model)               
```

Comment:
lets make use of CLI as much as possible? may be add asset_upate function similar to asset_create?

Reply:
Ignore this comment